### PR TITLE
Fix typos & review translation

### PR DESCRIPTION
**Pull request by atefBB**
The translation need re-`review` !

### CODE REVIEW

**Comment by atefBB** [comment]
@3imed-jaberi @Aissaoui-Ahmed Need your reviews

File: content/docs/thinking-in-react.md [hunk 1]
```diff
@@ -59,91 +59,88 @@ prev: composition-vs-inheritance.html
...
+مع نهاية هذه الخطوة، سيكون لديك مكتبة من المكونات القابلة لإعادة الاستخدام (reusable) التي تقوم بتصيير نموذج البيانات، المكونات ستحتوي فقط على دالات (`render()`) حيث أن هذه نسخة ثابتة من تطبيقك، المكون بأعلى التسلسل الهرمي (`FilterableProductTable`) سيحصل علي نموذج البيانات  بصفته خاصية (prop)، إذا قمت بعمل تغيير في نموذج البيانات وقمت باستدعاء الدالة (`ReactDOM.render()`) مرة أخرى فإن واجهة المستخدم سيتم تحديثها، من السهل رؤية كيف يتم تحديث واجهة المستخدم وأين تحدث التغييرات فلا يوجد شيء معقد، طريقة تدفق البيانات في اتجاه واحد (**one-way data flow**) الخاصة بـ (React) وتدعي أيضاَ بـ (*one-way binding*) تحافظ على كل شيء سريع ووحدة (modular) واحدة.
```
Comment (Aissaoui-Ahmed) [inline content/docs/thinking-in-react.md:73]:
```suggestion
مع نهاية هذه الخطوة، سيكون لديك مكتبة من المكونات القابلة لإعادة الاستخدام (reusable) التي تقوم بتصيير نموذج البيانات، المكونات ستحتوي فقط على دالات (`render()`) حيث أن هذه نسخة ثابتة من تطبيقك، المكون بأعلى التسلسل الهرمي (`FilterableProductTable`) سيحصل علي نموذج البيانات بصفته خاصية (prop)، إذا قمت بعمل تغيير في نموذج البيانات وقمت باستدعاء الدالة (`ReactDOM.render()`) مرة أخرى فإن واجهة المستخدم سيتم تحديثها، من السهل رؤية كيف يتم تحديث واجهة المستخدم وأين تحدث التغييرات فلا يوجد شيء معقد، طريقة تدفق البيانات في اتجاه واحد (**one-way data flow**) الخاصة بـ (React) وتدعي أيضاَ بـ (*one-way binding*) تحافظ على كل شيء سريع ووحدة (modular) واحدة.
```

File: content/docs/thinking-in-react.md [hunk 2]
```diff
@@ -59,91 +59,88 @@ prev: composition-vs-inheritance.html
...
-  * نظرياَ من المنطقى أن تتواجد كلمة البحث وقيمة ال (checkbox) فى المكون `FilterableProductTable`.
+  * نظرياَ من المنطقي أن تتواجد كلمة البحث وقيمة الـ (checkbox) في المكون `FilterableProductTable`.
 
-رائع، لقد قررنا أن الحالة ستكون فى المكون `FilterableProductTable`، أولاَ أضف (instance property) `this.state = {filterText: '', inStockOnly: false}` لل `constructor` للمكون `FilterableProductTable`  لتكون الحالة الإبتدائية للتطبيق ثم مرر الحالتين `filterText` و `inStockOnly` للمكونين `ProductTable` و `SearchBar` كخاصيات (props) وفى النهاية، إستخدم هذه الخاصيات (props) لتنقيح صفوف المنتجات فى المكون `ProductTable` ووضع القيم لل (form fields) فى المكون `SearchBar`.
+رائع، لقد قررنا أن الحالة ستكون في المكون `FilterableProductTable`، أولاَ أضف (instance property) `this.state = {filterText: '', inStockOnly: false}` لل `constructor` للمكون `FilterableProductTable`  لتكون الحالة الإبتدائية للتطبيق ثم مرر الحالتين `filterText` و `inStockOnly` للمكونين `ProductTable` و `SearchBar` بصفتها خاصيات (props) وفي النهاية، استخدم هذه الخاصيات (props) لتنقيح صفوف المنتجات في المكون `ProductTable` وضع القيم للـ (form fields) في المكون `SearchBar`.
```
Comment (Aissaoui-Ahmed) [inline content/docs/thinking-in-react.md:126]:
```suggestion
رائع، لقد قررنا أن الحالة ستكون في المكون `FilterableProductTable`، أولاَ أضف (instance property) `this.state = {filterText: '', inStockOnly: false}` لل `constructor` للمكون `FilterableProductTable` 
 لتكون الحالة الإبتدائية للتطبيق ثم مرر الحالتين `filterText` و `inStockOnly` للمكونين `ProductTable` و `SearchBar` بصفتها خاصيات (props) وفي النهاية، استخدم هذه الخاصيات (props) لتنقيح صفوف المنتجات في المكون `ProductTable` وضع القيم للـ (form fields) في المكون `SearchBar`.
```

File: content/docs/thinking-in-react.md [hunk 3]
```diff
@@ -59,91 +59,88 @@ prev: composition-vs-inheritance.html
...
+### نبذة مختصرة: الخاصية (Props) مقابل الحالة (State){#a-brief-interlude-props-vs-state}
 
-هناك نوعان من نماذج البيانات في (React): الحالة (state) و الخاصية (props)، من المهم فهم الأختلاف بين الأثنين، تصفح الوثائق [the official React docs](/docs/state-and-lifecycle.html) إذا لم تكن متأكد من الأختلاف. تصفح أيضا [الأسئلة الأكثر تكراراً: ما الفرق بين الحالة state والخاصيّات props؟](/docs/faq-state.html#what-is-the-difference-between-state-and-props)
+هناك نوعان من نماذج البيانات في (React): الحالة (state) والخاصية (props)، من المهم فهم الاختلاف بين الاثنين، تصفح الوثائق [the official React docs](/docs/state-and-lifecycle.html) إذا لم تكن متأكدا من الاختلاف. تصفح أيضا [الأسئلة الأكثر تكراراً: ما الفرق بين الحالة state والخاصيّات props؟](/docs/faq-state.html#what-is-the-difference-between-state-and-props)
 
-##  الخطوة الثالثة: تحديد الحد الأدني (ولكن المكتمل) الممثل لحالة (state) واجهة المستخدم{#step-3-identify-the-minimal-but-complete-representation-of-ui-state}
+##  الخطوة الثالثة: تحديد الحد الأدنى (ولكن المكتمل) الممثل لحالة واجهة المستخدم{#step-3-identify-the-minimal-but-complete-representation-of-ui-state}
```
Comment (Aissaoui-Ahmed) [inline content/docs/thinking-in-react.md:81]:
```suggestion
## الخطوة الثالثة: تحديد الحد الأدنى (ولكن المكتمل) الممثل لحالة واجهة المستخدم{#step-3-identify-the-minimal-but-complete-representation-of-ui-state}
```

File: content/docs/thinking-in-react.md [hunk 4]
```diff
@@ -35,7 +35,7 @@ prev: composition-vs-inheritance.html
...
 
-ولكن كيف تعرف ما يجب أن تحدده بصفته مكون؟ فقط استخدم نفس الأساليب أثناء أخذ قرار بإنشاء دالة (Function) أو كائن (Object) جديد، أسلوب "مبدأ المسئولية الأحادية" هو مثال على ذلك [single responsibility principle](https://ar.wikipedia.org/wiki/%D9%85%D8%A8%D8%AF%D8%A3_%D8%A7%D9%84%D9%85%D9%87%D9%85%D8%A9_%D8%A7%D9%84%D9%88%D8%A7%D8%AD%D8%AF%D8%A9) وهي أن المكون (component) بشكل مثالي يجب أن يكون مسئولا عن فعل شيء واحد فقط وإذا بدأ في التنامي يجب تقسيمه إلى مكونات فرعية (subcomponent) أصغر.
+ولكن كيف تعرف ما يجب أن تحدده بصفته مكونا؟ فقط استخدم نفس الأساليب أثناء أخذ قرار بإنشاء دالة (Function) أو كائن (Object) جديد، أسلوب "مبدأ المسئولية الأحادية" هو مثال على ذلك [single responsibility principle](https://ar.wikipedia.org/wiki/%D9%85%D8%A8%D8%AF%D8%A3_%D8%A7%D9%84%D9%85%D9%87%D9%85%D8%A9_%D8%A7%D9%84%D9%88%D8%A7%D8%AD%D8%AF%D8%A9) وهي أن المكون (component) بشكل مثالي يجب أن يكون مسئولا عن فعل شيء واحد فقط، وإذا بدأ في التنامي يجب تقسيمه إلى مكونات فرعية (subcomponent) أصغر.
```
Comment (Aissaoui-Ahmed) [inline content/docs/thinking-in-react.md:38]:
```suggestion
ولكن كيف تعرف ما يجب أن تحدده بصفته مكونا؟ فقط استخدم نفس الأساليب أثناء أخذ قرار بإنشاء دالة (Function) أو كائن (Object) جديد، أسلوب "مبدأ المسئولية الأحادية" هو مثال على ذلك [مبدأ المهمة الواحدة](https://ar.wikipedia.org/wiki/%D9%85%D8%A8%D8%AF%D8%A3_%D8%A7%D9%84%D9%85%D9%87%D9%85%D8%A9_%D8%A7%D9%84%D9%88%D8%A7%D8%AD%D8%AF%D8%A9) وهي أن المكون (component) بشكل مثالي يجب أن يكون مسئولا عن فعل شيء واحد فقط، وإذا بدأ في التنامي يجب تقسيمه إلى مكونات فرعية (subcomponent) أصغر.
```

**Review by Aissaoui-Ahmed — CHANGES_REQUESTED** [review]
Remove double space

**Comment by atefBB** [comment]
@Aissaoui-Ahmed DONE!

**Review by Aissaoui-Ahmed — APPROVED** [review]

**Pull request merged**